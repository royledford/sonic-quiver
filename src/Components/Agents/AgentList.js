import React, { Component } from 'react'
import PropTypes from 'prop-types'
import AgentSmall from './AgentSmall'
import './AgentList.css'

export default class AgentList extends Component {
  static PropTypes = {
    data: PropTypes.array.isRequired,
  }
  static defaultProps = {
    data: [],
  }
  render() {
    const { data } = this.props
    const listItems = data.map(agent => {
      return (
        <AgentSmall
          key={agent.id}
          id={agent.id}
          location={agent.shortCountry}
          companyName={agent.name}
          agentName={agent.agentName}
          onMouseEnter={this.props.onMouseEnter}
        />
      )
    })
    return (
      <div className="agentlist">
        <ul>{listItems}</ul>
      </div>
    )
  }
}
