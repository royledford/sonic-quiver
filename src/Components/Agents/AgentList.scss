.agentlist {
  position: relative;
  width: 636px;
  // height: 100%;
  overflow: hidden;
  overflow-y: scroll;
  direction:rtl;
}