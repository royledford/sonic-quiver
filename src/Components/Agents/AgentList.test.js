import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'

import AgentList from './AgentList'
import renderer from 'react-test-renderer'

it('AgentSmall renders', () => {
  const tree = renderer
    .create(
      <AgentList
        data={[
          {
            id: 1,
            country: 'United States',
            shortCountry: 'USA',
            sortString: 'AAAA',
            name: '5 Alarm Music',
            address1: '35 West Dayton St.',
            address2: 'Pasadena, CA 91105',
            address3: '',
            agentName: 'Cassie Lord',
            phone: '(626) 304-1698',
            fax: '(626) 795-2058',
            email: 'cassie@5alarmmusic.com',
            url: 'http://5alarmmusic.com/',
            urlDisplay: '5alarmmusic.com',
            sonicUrl: 'http://5alarmmusic.com/music/sonic-quiver/',
          },
        ]}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
