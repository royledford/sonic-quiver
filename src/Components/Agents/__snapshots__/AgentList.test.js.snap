// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentSmall renders 1`] = `
<div
  className="agentlist"
>
  <ul>
    <div
      className="agentsmall-wrap"
      onMouseEnter={[Function]}
    >
      <h4
        className="country"
      >
        USA
      </h4>
      <hr
        className="agentsmall-rule"
      />
      <p
        className="agentsmall-copy"
      >
        5 Alarm Music
         - 
        Cassie Lord
      </p>
      <div
        className="agentsmall-mobile"
      >
        <p>
          <a
            href="mailto:cassie@5alarmmusic.com"
          >
            cassie@5alarmmusic.com
          </a>
        </p>
        <a
          href="http://5alarmmusic.com/music/sonic-quiver/"
          rel="noopener noreferrer"
          target="_blank"
        >
          5alarmmusic.com
        </a>
        <div
          className="agentsmall-copy"
        >
          TEL: 
          (626) 304-1698
          <br />
          FAX: (626) 795-2058
        </div>
        <div
          className="agentsmall-copy"
        >
          35 West Dayton St.
          <br />
          Pasadena, CA 91105
          <br />
          
          <br />
        </div>
      </div>
    </div>
  </ul>
</div>
`;
